packages/ipa-certified-attributes-importer/src/index.ts

```diff
@@ -0,0 +1,467 @@
+    )
+  );
+
+  const tenantSeedIndex = new Map(
+    tenantSeed.map((t) => [
+      toKey({ origin: t.origin, value: t.originId }),
+      new Set(
+        t.attributes.map((a) => toKey({ origin: a.origin, value: a.code }))
+      ),
+    ])
+  );
+
+  const certifiedAttribute = new Map(
+    platformAttributes
+      .filter((a) => a.kind === "Certified" && a.origin && a.code)
+      .map((a) => [a.id, a])
+  );
+
+  const canBeRevoked = (
+    attribute: {
+      origin: string;
+      code: string;
+    },
+    tenantExternalId: { origin: string; value: string }
+  ): boolean => {
+    const externalId = { origin: attribute.origin, value: attribute.code };
+
+    if (attribute.origin !== ORIGIN_IPA) {
+      return false;
+    }
+
+    const registryAttributes = tenantSeedIndex.get(toKey(tenantExternalId));
+    if (!registryAttributes) {
+      return false;
+    }
+
+    if (registryAttributes.has(toKey(externalId))) {
+      return false;
+    }
+
+    return !indexFromOpenData.has(toKey(externalId));
+  };
+
+  return platformTenants.flatMap((t) =>
+    t.attributes
+      // eslint-disable-next-line sonarjs/no-identical-functions
+      .map((a) => {
+        const withRevocation = match(a)
+          .with({ type: "PersistentCertifiedAttribute" }, (certified) => ({
+            ...a,
+            revocationTimestamp: certified.revocationTimestamp,
+          }))
+          .otherwise((_) => undefined);
+
+        if (withRevocation) {
+          const attribute = certifiedAttribute.get(withRevocation.id);
+
+          if (attribute) {
+            return {
+              ...attribute,
+              revocationTimestamp: withRevocation.revocationTimestamp,
+            };
+          }
+        }
+
+        return undefined;
+      })
+      .filter(
+        (a) =>
+          a !== undefined &&
+          a.revocationTimestamp === undefined &&
+          a.origin &&
+          a.code &&
+          canBeRevoked(
+            {
+              origin: a.origin,
+              code: a.code,
+            },
+            t.externalId
+          )
+      )
+      .map((a) => ({
+        tOrigin: t.externalId.origin,
+        tExtenalId: t.externalId.value,
+        aOrigin: a?.origin as string,
+        aCode: a?.code as string,
+      }))
+  );
+}
+
+async function revokeAttributes(
+  attributesToRevoke: Array<{
+    tOrigin: string;
+    tExtenalId: string;
+    aOrigin: string;
+    aCode: string;
+  }>,
+  headers: Header
+): Promise<void> {
+  const tenantClient = tenantApi.createInternalApiClient(
+    config.tenantProcessUrl
+  );
+
+  for (const a of attributesToRevoke) {
+    await tenantClient.internalRevokeCertifiedAttribute(undefined, {
+      params: {
+        tOrigin: a.tOrigin,
+        tExternalId: a.tExtenalId,
+        aOrigin: a.aOrigin,
+        aExternalId: a.aCode,
+      },
+      headers,
+    });
+  }
+}
+
+async function getHeader(
+  refreshableToken: RefreshableInteropToken,
+  correlationId: string
```

Comment:
If you merge main you could:
```suggestion
  correlationId: CorrelationId
```